apps/teams-test-app/src/components/MediaAPIs.tsx

```diff
@@ -69,19 +107,47 @@ const GetMedia = (): React.ReactElement =>
     onClick: {
       // eslint-disable-next-line @typescript-eslint/no-empty-function
       validateInput: () => {},
-      submit: async (input, setResult) => {
-        const medias = await media.selectMedia(input);
-        const mediaItem: media.Media = medias[0] as media.Media;
-        const blob = await mediaItem.getMedia();
-        const reader = new FileReader();
-        reader.readAsDataURL(blob);
-        reader.onloadend = () => {
-          if (reader.result) {
-            // eslint-disable-next-line  @typescript-eslint/no-explicit-any
-            setResult('Received Blob (length: ' + (reader.result as any).length + ')');
-          }
-        };
-        return 'media.getMedia()' + noHostSdkMsg;
+      submit: {
+        withPromise: async (input, setResult) => {
+          const medias = await media.selectMedia(input);
+          const mediaItem: media.Media = medias[0] as media.Media;
+          const blob = await mediaItem.getMedia();
+          const reader = new FileReader();
+          reader.readAsDataURL(blob);
+          reader.onloadend = () => {
+            if (reader.result) {
+              // eslint-disable-next-line  @typescript-eslint/no-explicit-any
+              setResult('Received Blob (length: ' + (reader.result as any).length + ')');
+            }
+          };
+          return 'media.getMedia()' + noHostSdkMsg;
+        },
+        withCallback: (input, setResult) => {
+          const getMediaCallback = (error: SdkError, blob: Blob): void => {
+            if (error) {
+              setResult(JSON.stringify(error));
+            } else {
+              const reader = new FileReader();
+              reader.readAsDataURL(blob);
+              reader.onloadend = () => {
+                if (reader.result) {
+                  // eslint-disable-next-line  @typescript-eslint/no-explicit-any
+                  setResult('Received Blob (length: ' + (reader.result as any).length + ')');
```

Comment:
I feel like this could be extracted and then reused, since it is used in both withPromise and withCallback. I'll experiment with it.